yarn/default.nix

```diff
@@ -1,10 +1,19 @@
 { callPackage }:
 
 rec {
-  yarn = callPackage ./yarn.nix {};
-  yarn-patched = yarn.override {
-    applyBuiltinPatches = true;
+  yarn-unwrapped = callPackage ./yarn.nix {};
+
+  wrapYarnWith = { yarn, ... } @ extraArgs: callPackage ./wrapper.nix ({
+    inherit yarn;
+  } // extraArgs);
+
+  yarn = wrapYarnWith {
+    yarn = yarn-unwrapped;
```

Comment:
This an awkward part, for reference upstream has
```nix
{
  binutils = wrapBintoolsWith {
    bintools = binutils-unwrapped;
  };
}
```

Reply:
`yarn-unwrapped` here is just a js file without a shebang. My impression was that the `-unwrapped` stuff in nixpkgs are actually usable derivations with a `bin/` folder.

Reply:
That's a fair point. I was finding it quite awkward that the wrapper also contained the non-trivial build of yarn (since it includes patches). Maybe there's an alternate suffix we could use?

Reply:
I went with calling it `yarn-js` everywhere (as the attrname version of `yarn.js` but left the `wrapYarnWith` as-is, does that sounds good?

Reply:
That works for me.